File: Python/queue.py
class Queue(object):
    def __init__(self):
        """
        initialize your data structure here.
        """
        self.stack = []
        self.size = 0
        

    def push(self, x):
        """
        :type x: int
        :rtype: nothing
        """
        stk = []
        for _ in xrange(self.size):
            stk.append(self.stack.pop())
        self.stack.append(x)
        for _ in xrange(self.size):
            self.stack.append(stk.pop())
        self.size += 1
        

    def pop(self):
        """
        :rtype: nothing
        """
        self.stack.pop()
        self.size-=1
        

    def peek(self):
        """
        :rtype: int
        """
        return self.stack[-1]
        
        

    def empty(self):
        """
        :rtype: bool
        """
        return self.size == 0
        

File: src/Queue.js
/**
 * @constructor
 */
var Queue = function() {
    this.stack = [];
    this.tmp = [];
};

/**
 * @param {number} x
 * @returns {void}
 */
Queue.prototype.push = function(x) {
    this.stack.push(x);
};

/**
 * @returns {void}
 */
Queue.prototype.pop = function() {
    while(this.stack.length > 1){
        this.tmp.push(this.stack.pop());
    }
    this.stack.pop();
    while(this.tmp.length > 0){
        this.stack.push(this.tmp.pop());
    }
};

/**
 * @returns {number}
 */
Queue.prototype.peek = function() {
    while(this.stack.length > 1){
        this.tmp.push(this.stack.pop());
    }
    var ele = this.stack.pop();
    this.tmp.push(ele);
    while(this.tmp.length > 0){
        this.stack.push(this.tmp.pop());
    }
    return ele;
};

/**
 * @returns {boolean}
 */
Queue.prototype.empty = function() {
    if(this.stack === null) return true;
    return this.stack.length === 0;
    
};